modules/cudafilters/src/cuda/wavelet_matrix_2d.cuh

```diff
@@ -0,0 +1,1011 @@
+#ifndef __OPENCV_WAVELET_MATRIX_2D_CUH__
```

Comment:
Please add OpenCV license header in added files: https://github.com/opencv/opencv/wiki/Coding_Style_Guide#file-structure

```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.
```

If this code is borrowed from somewhere then please add original license.